indico/modules/receipts/util.py

```diff
@@ -145,22 +148,58 @@ def compile_jinja_code(code: str, template_context: dict, *, use_stack: bool = F
         raise UnprocessableEntity(e)
 
 
-def sandboxed_url_fetcher(event: Event, allow_event_logo: bool = False) -> t.Callable[[str], dict]:
+def sandboxed_url_fetcher(event: Event, allow_event_images: bool = False) -> t.Callable[[str], dict]:
     """Fetch also "event-local" URLs.
 
     More info on fetchers: https://doc.courtbouillon.org/weasyprint/stable/first_steps.html#url-fetchers
     """
     allow_external_urls = receipts_settings.get('allow_external_urls')
 
     def _fetcher(url: str) -> dict:
-        if allow_event_logo and url == 'event://logo':
-            return {
-                'mime_type': event.logo_metadata['content_type'],
-                'string': event.logo
-            }
+        url_data = urlparse(url)
+        if allow_event_images and url_data.scheme == 'event':
+            if url_data.netloc == 'logo':
+                return {
+                    'mime_type': event.logo_metadata['content_type'],
+                    'string': event.logo
+                }
+            if url_data.netloc == 'placeholder':
+                placeholder_image = Path(current_app.root_path) / 'web' / 'static' / 'images' / 'placeholder_image.svg'
+                return {
+                    'mime_type': 'image/svg+xml',
+                    'string': placeholder_image.read_text()
+                }
+            try:
+                img_id = int(url_data.path[1:])
+            except ValueError:
+                raise ValueError('Invalid event-local image reference')
+            image = None
+            if url_data.netloc == 'images':
+                image = event.layout_images.filter_by(id=img_id).first()
+            elif url_data.netloc == 'attachments' and (attachment := get_event_attachment_images(event).get(img_id)):
+                image = attachment.file
+            if not image:
+                raise ValueError('Invalid event-local image reference')
+            try:
+                with image.open() as f:
+                    if image.content_type == 'image/svg+xml':
+                        return {
+                            'mime_type': 'image/svg+xml',
+                            'string': f.read()
+                        }
+                    with Image.open(f) as pic:
+                        if pic.format.lower() not in {'jpeg', 'png', 'gif', 'webp'}:
+                            raise ValueError('Unsupported image format')
+                        output = BytesIO()
+                        pic.save(output, format=pic.format)
```

Comment:
I suggest always converting to PNG here. You don't want a JPEG-to-JPEG roundtrip for example (that degrades the quality)